controller_interface/test/test_controller_with_options.cpp

```diff
@@ -55,6 +58,98 @@ TEST(ControllerWithOption, init_with_overrides)
+  FriendControllerWithOptions controller;
+  const std::string params_file_path = ament_index_cpp::get_package_prefix("controller_interface") +
+                                       "/test/test_controller_node_options.yaml";
+  std::cerr << params_file_path << std::endl;
+  auto controller_node_options = controller.define_custom_node_options();
+  controller_node_options.arguments({"--ros-args", "--params-file", params_file_path});
+  EXPECT_EQ(
+    controller.init("controller_name", "", 50.0, "", controller_node_options),
+    controller_interface::return_type::OK);
+  // checks that the node options have been updated
+  const auto & node_options = controller.get_node()->get_node_options();
+  EXPECT_TRUE(node_options.allow_undeclared_parameters());
+  EXPECT_TRUE(node_options.automatically_declare_parameters_from_overrides());
+  EXPECT_EQ(3lu, node_options.arguments().size());
+  // checks that the parameters have been correctly processed
+  EXPECT_EQ(controller.params.size(), 3u);
+  EXPECT_EQ(controller.params["parameter1"], 20.0);
+  EXPECT_EQ(controller.params["parameter2"], 23.14);
+  EXPECT_EQ(controller.params["parameter3"], -52.323);
+  bool use_sim_time(true);
+  controller.get_node()->get_parameter_or("use_sim_time", use_sim_time, false);
+  ASSERT_FALSE(use_sim_time);
+  rclcpp::shutdown();
+}
+
+TEST(
+  ControllerWithOption, init_with_node_options_arguments_parameters_file_and_override_command_line)
+{
+  char const * const argv[] = {""};
+  int argc = arrlen(argv);
+  rclcpp::init(argc, argv);
+  // creates the controller
+  FriendControllerWithOptions controller;
+  const std::string params_file_path = ament_index_cpp::get_package_prefix("controller_interface") +
+                                       "/test/test_controller_node_options.yaml";
+  std::cerr << params_file_path << std::endl;
+  auto controller_node_options = controller.define_custom_node_options();
+  controller_node_options.arguments(
+    {"--ros-args", "--params-file", params_file_path, "-p", "parameter_list.parameter1:=562.785",
+     "-p", "use_sim_time:=true"});
+  EXPECT_EQ(
+    controller.init("controller_name", "", 50.0, "", controller_node_options),
+    controller_interface::return_type::OK);
+  // checks that the node options have been updated
+  const auto & node_options = controller.get_node()->get_node_options();
+  EXPECT_TRUE(node_options.allow_undeclared_parameters());
+  EXPECT_TRUE(node_options.automatically_declare_parameters_from_overrides());
+  EXPECT_EQ(7lu, node_options.arguments().size());
+  // checks that the parameters have been correctly processed
+  EXPECT_EQ(controller.params.size(), 3u);
+  EXPECT_EQ(controller.params["parameter1"], 562.785);
+  EXPECT_EQ(controller.params["parameter2"], 23.14);
+  EXPECT_EQ(controller.params["parameter3"], -52.323);
+  bool use_sim_time(false);
+  controller.get_node()->get_parameter_or("use_sim_time", use_sim_time, false);
+  ASSERT_TRUE(use_sim_time);
+  rclcpp::shutdown();
+}
```

Comment:
@destogl @bmagyar @christophfroehlich These  should give you guys an idea on how we are going to tackle in the CM with the other PR